### ISSUE TITLE
[bug] [0.8.0] escaped $ are not respected

### ISSUE THREAD

**Issue by laur89** [foundweekends/giter8#288]
Parser interprets escaped $-signs as placeholders, given there are at least 2 of them on a single line. This causes 2 problems
- user is asked for values for these;
- g8 fails if the pseudo-placeholder doesn't adhere to required format (eg when it contains a dot)

Note that final expanded file is still OK - no replacements take place.

0.8.0:

    g8 https://github.com/laur89/unfiltered.g8.git
    This template generates an Unfiltered project

    name [My Web Project]: 
    unfiltered_version [0.9.1]: 
    description [This template generates an Unfiltered project]: 
    scala_version [2.12.1]: 
    version [0.1.0-SNAPSHOT]: 
    SHOULD_NOT_ASK_BUT_WONT_EXPAND_SO_ITS_KINDA_OK \ []: 
    THIS_ALSO_FAILS_BECAUSE_IT_HAS_A_DOT_IN_IT. \ []: 
    cannot have '.' in attribute names

0.7.2 (which is ok):

    g8 https://github.com/laur89/unfiltered.g8.git

    This template generates an Unfiltered project 

    name [My Web Project]: 
    version [0.1.0-SNAPSHOT]: 
    scala_version [2.12.1]: 
    unfiltered_version [0.9.1]: 

    Template applied in ./my-web-project

The failing template is [here](https://github.com/laur89/unfiltered.g8/blob/master/src/main/g8/build.sbt); contents for future reference:

    # both of these are erroneously asked from the user (but final result is not expanded);
    # second one causes result to fail because of the dot
    shouldNotExpand1="\$SHOULD_NOT_ASK_BUT_WONT_EXPAND_SO_ITS_KINDA_OK \$THIS_IS_ALSO_NEEDED"
    shouldNotExpand2="\$THIS_ALSO_FAILS_BECAUSE_IT_HAS_A_DOT_IN_IT. \$THIS_IS_ALSO_NEEDED"

**Comment by eed3si9n** [comment]
To clarify, one of your files has:

```scala
shouldNotExpand1="\$SHOULD_NOT_ASK_BUT_WONT_EXPAND_SO_ITS_KINDA_OK \$THIS_IS_ALSO_NEEDED"
shouldNotExpand2="\$THIS_ALSO_FAILS_BECAUSE_IT_HAS_A_DOT_IN_IT. \$THIS_IS_ALSO_NEEDED"
```

in Giter8 0.8.0 it started picking these up as template variables even though they are escaped.

**Comment by laur89** [comment]
So it's a feature?

**Comment by eed3si9n** [comment]
I think it's a bug. @foxmk?

**Comment by foxmk** [comment]
Yes, this is a regression. I'll check escaping support.

**Comment by DWvanGeest** [comment]
Can confirm I'm seeing this too, it's kind of a deal breaker when you're generating Scala code with plenty of string interpolation....

**Comment by imarios** [comment]
Hey guys, anyone looking at this?

**Comment by lachatak** [comment]
It is show stopper for us to change to 0.8.0 from 0.7.2 because of this regression. Any plans?

**Comment by eed3si9n** [comment]
I "fixed" this by rolling back 0.8.0 changes. Stay tuned for 0.9.0.

```
$ g8 https://github.com/laur89/unfiltered.g8.git
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0

This template generates an Unfiltered project

name [My Web Project]: foo
version [0.1.0-SNAPSHOT]:
scala_version [2.12.1]:
unfiltered_version [0.9.1]:

Template applied in /Users/xxx/./foo
```

**Comment by fabriziofortino** [comment]
@eed3si9n I am getting the following error when I try to apply a template with the latest from master:

```
Exiting due to error in the template
File: /var/folders/22/g7d85p4j2cx7bj3sf2mb0tg489wxn0/T/giter8-1555432061776410/akka-http/src/main/g8/src/main/scala/$package$/routes/package.scala, 50:40: 'could' came as a complete surprise to me
```

Here is the line that causes it:
```scala
logger.error(e, s"Request to $uri could not be handled normally")
```
